### ISSUE TITLE
Invalid Hijrah day of month: 30

### ISSUE THREAD

**Issue by atifaziz1** [msarhan/ummalqura-calendar#35]
I am facing issue that hijrah date is invalid.. This error is exist in both version 2.0.1 and 2.0.2. 

Any help.?  @msarhan @alhazmy13 
<img width="1276" alt="Screenshot 2022-11-24 at 11 31 00 AM" src="https://user-images.githubusercontent.com/37400835/203732359-e6426178-323b-4258-a0c7-ad9782801e18.png">

Error Screen shot is attached.

**Comment by alhazmy13** [comment]
@atifaziz1 I think you are parsing the date in the wrong way; you cannot set the month to 30.
Can you share your code?

**Comment by atifaziz1** [comment]
@alhazmy13 Thanks for your quick response.

Here is my code

UmmalquraCalendar now = new UmmalquraCalendar();

HijriDatePickerDialog hijriDatePickerDialog = HijriDatePickerDialog.newInstance(listener,now.get(UmmalquraCalendar.YEAR), now.get(UmmalquraCalendar.MONTH), now.get(UmmalquraCalendar.DAY_OF_MONTH));`

**Comment by atifaziz1** [comment]
@alhazmy13  If i do 

`UmmalquraCalendar now = new UmmalquraCalendar();`
`HijriDatePickerDialog hijriDatePickerDialog = HijriDatePickerDialog.newInstance(listener,  now.get(Calendar.YEAR), now.get(Calendar.MONTH),now.get(Calendar.DAY_OF_MONTH));`

I am getting ` com.github.msarhan.ummalqura.calendar.DateTimeException: Invalid Hijrah day of month: 31` error

**Comment by atifaziz1** [comment]
@msarhan please help required. It's urgent.

**Comment by alhazmy13** [comment]
@atifaziz1 I believe this related to [HijriDatePicker](https://github.com/alhazmy13/HijriDatePicker) it could be using an old version of `UmmalquraCalendar` or something like that, ill dig deep on the library to find the root case of this issue

**Comment by atifaziz1** [comment]
@alhazmy13 I hope you will fix this issue on priority bases.
Thanks.

**Comment by alhazmy13** [comment]
Yes sir @atifaziz1 , you are using the library in the wrong way, please update your code to below:
```
UmmalquraCalendar now = new UmmalquraCalendar();
HijriDatePickerDialog dpd = HijriDatePickerDialog.newInstance(
		  this,
		  now.get(UmmalquraCalendar.YEAR),
		  now.get(UmmalquraCalendar.MONTH),
		  now.get(UmmalquraCalendar.DAY_OF_MONTH));
dpd.show(getFragmentManager(), "HijriDatePickerDialog");
```

more details: https://github.com/alhazmy13/HijriDatePicker#create-a-hijridatepickerdialog

**Comment by atifaziz1** [comment]
Dear @alhazmy13 !

Please look at my first comment, i am writing the code same as you mentioned.

```
UmmalquraCalendar now = new UmmalquraCalendar();

HijriDatePickerDialog hijriDatePickerDialog = HijriDatePickerDialog.newInstance(listener,
now.get(UmmalquraCalendar.YEAR), 
now.get(UmmalquraCalendar.MONTH),
 now.get(UmmalquraCalendar.DAY_OF_MONTH));`
```

**Comment by AhmedElsayedTaha** [comment]
@atifaziz1  Did you fixed it ? I'm stuck at this error

**Comment by AhmedElsayedTaha** [comment]
@msarhan  @alhazmy13  Hello , I have the same exception when I choose day 30 from any month then choose day 29 from another month I get  this exception . I use version 2.0.2
 
 Here is my code 
 
```
 private fun showHijriDatePicker(
        editText: TextInputEditText,
        isFromDatePicker: Boolean = false
    ) {
        val datePicker = HijriDatePickerDialog.newInstance(
            { _, year, monthOfYear, dayOfMonth ->
                when (isFromDatePicker) {
                    true -> {
                        fromDateValue = formatHijriDate(year, monthOfYear, dayOfMonth)
                    }
                    false -> {
                        toDateValue = formatHijriDate(year, monthOfYear, dayOfMonth)
                    }
                }
                editText.setText(formatHijriDate(year, monthOfYear, dayOfMonth))
            },
            ummalquraCalendar.get(UmmalquraCalendar.YEAR),
            ummalquraCalendar.get(UmmalquraCalendar.MONTH),
            ummalquraCalendar.get(UmmalquraCalendar.DAY_OF_MONTH)
        )
}
```

![image](https://github.com/msarhan/ummalqura-calendar/assets/49143167/32aa454b-4ca2-4dac-8d3e-dfb6000b3747)

**Comment by MouazSalah** [comment]
Hi all 👋

I just opened [#39 – Crash in HijriDatePickerDialog When Switching From 29-Day Month to 30-Day Month](https://github.com/msarhan/ummalqura-calendar/issues/39), and it looks like the exact same problem reported here —
DateTimeException: Invalid Hijrah day of month: 30 when moving from a 29-day month to a 30-day month.

@msarhan  @alhazmy13 @atifaziz1 @AhmedElsayedTaha 
Did anyone manage to fix this in the library or find a reliable workaround?
I’m still facing it in the latest version.
